test/apiserver_organization_test.go

```diff
@@ -135,22 +117,212 @@ modfor:
+	t.Cleanup(cl.CleanUpFunc(ctx))
+	dcl, err := dynamic.NewForConfig(cfg)
+	require.NoError(t, err)
+	wi, err := dcl.Resource(gvr).Watch(ctx, metav1.ListOptions{
+		LabelSelector: "test-name=" + t.Name(),
+	})
+	require.NoError(t, err)
+	globalEventTraced := events.NewTracer(wi)
+	t.Cleanup(globalEventTraced.TestCleanupFunc(t))
+
+	owner := rbacv1.Subject{
+		Kind:     "User",
+		APIGroup: "rbac.authorization.k8s.io",
+		Name:     "kubernetes-admin",
+	}
+	testCase := []*TestVisibleFilteringTestCase{
+		{
+			Name: "user",
+			Subject: rbacv1.Subject{
+				Kind:     rbacv1.UserKind,
+				APIGroup: rbacv1.GroupName,
+				Name:     "user",
+			},
+			ImpersonationConfig: rest.ImpersonationConfig{
+				UserName: "user",
+			},
+		},
+		{
+			Name: "group",
+			Subject: rbacv1.Subject{
+				Kind:     rbacv1.GroupKind,
+				APIGroup: rbacv1.GroupName,
+				Name:     "group",
+			},
+			ImpersonationConfig: rest.ImpersonationConfig{
+				UserName: "lala",
+				// without "system:authenticated" things are breaking...cannot RESTMapper doesn't function
+				Groups: []string{"system:authenticated", "group"},
+			},
+		},
+		{
+			Name: "sa",
+			Subject: rbacv1.Subject{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      "default",
+				Namespace: "default",
+			},
+			ImpersonationConfig: rest.ImpersonationConfig{
+				UserName: "system:serviceaccount:default:default",
+			},
+		},
+	}
+	t.Log("ensuring cluster RBAC")
+	ensureClusterRBAC(t, ctx, cl, testCase)
+
+	t.Log("creating orgs")
+	for _, tc := range testCase {
+		cfg, err := ctrl.GetConfig()
+		require.NoError(t, err)
+		cfg.Impersonate = tc.ImpersonationConfig
+		cfg.UserAgent = t.Name() + "/" + tc.Name
+		tc.Client = testutil.NewRecordingClient(t, cfg, testScheme, testutil.CleanUpStrategy(cleanUpStrategy))
+		t.Cleanup(tc.Client.CleanUpFunc(ctx))
+		dcl, err := dynamic.NewForConfig(cfg)
+		require.NoError(t, err)
+		wi, err := dcl.Resource(gvr).Watch(ctx, metav1.ListOptions{
+			LabelSelector: "test-name=" + t.Name(),
+		})
+		require.NoError(t, err)
+		tc.tracer = events.NewTracer(wi, events.IsObjectName("test-"+tc.Name))
+		t.Cleanup(tc.tracer.TestCleanupFunc(t))
+		tc.Org = &apiserverv1alpha1.Organization{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-" + tc.Name,
+				Labels: map[string]string{
+					"test-name": t.Name(),
+				},
+			},
+			Spec: storagev1alpha1.OrganizationSpec{
+				Metadata: &storagev1alpha1.OrganizationMetadata{
+					DisplayName: "test",
+					Description: "desc",
+				},
+				Owners: []rbacv1.Subject{owner},
+			},
+		}
+		require.Error(t, tc.Client.Create(ctx, tc.Org), "I cannot create an organization which I'm not owner of")
+		require.NoError(t, cl.Create(ctx, tc.Org))
+		assert.NoError(t, globalEventTraced.WaitUntil(ctx, events.AllOf(
+			events.IsType(watch.Added),
+			events.IsObjectName(tc.Org.Name),
+		)))
+		require.NoError(t, testutil.WaitUntilReady(ctx, cl, tc.Org))
+		rb := &rbacv1.RoleBinding{
```

Comment:
Since now we have organizationRoleTemplate controller to create role in organization namespace, you can just create RoleBindings to bind the role which created by these two templates: https://github.com/kubermatic/bulward/blob/master/pkg/templates/organizationrole.go

Reply:
it's irrelevant to which role I bind as long as I'm in the organization namespace I become member of it.

Reply:
but here you are admin so you can bind it to the ClusterRole, but in the real scenario, owner doens't have permission to create the rolebinding to bind member to the ClusterRole which escalates his/her permission.

Reply:
The owner is the cluster-admin, in all scenarios owner is the cluster-admin. In this test case, I'm testing how member view filtering works, not whether I'm having cluster-admin privileges in the real scenario or not.

Reply:
> The owner is the cluster-admin, in all scenarios owner is the cluster-admin.

I disagree with this. Owner only has permission to:
1. manage Projects
2. manage role/roleBindings in Organization namespace.
3. manage Organization (to be discussed)
Here you are creating RoleBinding with the client `cl` which has cluser-admin privileges, if you use impersonating Client of the Owner, I don't think you can create this RoleBinding.

Reply:
Well somebody gotta give users permissions to view the organizations in the first place, doesn't it?

How would you add arbitrary users to bulwarks which currently aren't part of it?